k8s/upgrade-job/src/upgrade/data_plane.rs

```diff
@@ -140,18 +142,69 @@ async fn verify_data_plane_pod_is_running(
     namespace: String,
     k8s_client: &KubeClientSet,
 ) -> Result<()> {
+    let io_engine_listparam = ListParams::default().labels(IO_ENGINE_LABEL);
+    let initial_io_engine_pod_list: ObjectList<Pod> = k8s_client
+        .pods_api()
+        .list(&io_engine_listparam)
+        .await
+        .context(ListPodsWithLabel {
+            label: IO_ENGINE_LABEL.to_string(),
+            namespace: namespace.clone(),
+        })?;
+
+    let mut io_engine_pod_uid = "";
+
+    for pod in initial_io_engine_pod_list.iter() {
+        let node = pod
+            .spec
+            .as_ref()
+            .ok_or(
+                EmptyPodSpec {
+                    name: pod.name_any(),
+                    namespace: namespace.clone(),
+                }
+                .build(),
+            )?
+            .node_name
+            .as_ref()
+            .ok_or(
+                EmptyPodNodeName {
+                    name: pod.name_any(),
+                    namespace: namespace.clone(),
+                }
+                .build(),
+            )?
+            .as_str();
+
+        // get pod uid running on this node
+        if node == node_name {
+            io_engine_pod_uid = pod.metadata.uid.as_ref().ok_or(
+                EmptyPodUid {
+                    name: pod.name_any(),
+                    namespace: namespace.clone(),
+                }
+                .build(),
+            )?;
```

Comment:
This can be done with the field-selector listparam where we can specify the criteria "spec.nodeName=<node-name>". Iterating over all of the Pods is not needed, the kubernetes API can bear this cost.